### PR TITLE
Publish nightly knative/build-pipeline releases to knative-nightly

### PR DESCRIPTION
**Pull request by adrcunha**
Part of knative/serving#2555.

Bonus: added documentation about the `/hack` dir and the release script.

### CODE REVIEW

File: hack/release.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+* `--version` Defines the version of the release, and must be in the form
+`X.Y.Z`, where X, Y and Z are numbers.
+* `--branch` Defines the branch in Knative Build Pipeline repository from which
+the release will be built. If not passed, the `master` branch at HEAD will be
+used. This branch must be created before the script is executed, and must be
+in the form `release-X.Y`, where X and Y must match the numbers used in the
+version passed in the `--version` flag. This flag has no effect unless
+`--version` is also passed.
+* `--release-notes` Points to a markdown file containing a description of the
```
Comment (bobcatfish) [inline hack/release.md]: do you have any recommendations on how we can get release notes going for this project? is there any automation around it in the other repos, or is it a file one creates manually? (asking b/c we need something like this for #72 :D)

File: hack/release.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+* `--tag-release`, `--notag-release` Tag (or not) the generated images
+with either `vYYYYMMDD-<commit_short_hash>` (for nightly releases) or
+`vX.Y.Z` for versioned releases. *For versioned releases, a tag is always
+added.*
+* `--publish`, `--nopublish` Whether the generated images should be published
+to a GCR, and the generated manifests written to a GCS bucket or not. If yes,
+the destination GCR is defined by the environment variable
+`$BUILD_PIPELINE_RELEASE_GCR` (defaults to `gcr.io/knative-nightly`) and the
+destination GCS bucket is defined by the environment variable
+`$BUILD_PIPELINE_RELEASE_GCS` (defaults to `knative-nightly/build`). If no,
```
[the remark this answers is not in the document]
Comment (adrcunha) [reply hack/release.md]: Good catch, fixed.

File: hack/release.md (hunk 1 above)
Comment (adrcunha) [reply hack/release.md]: Currently this is a manual step. We have the same need for other repos, https://github.com/knative/test-infra/issues/214 is the open issue for that.

**Comment by bobcatfish** [comment]
ugh sorry about that @adrcunha , that helm test has flaked out on us a few times, im gonna add an issue to the backlog to de-flake it

**Pull request merged**